package/MDAnalysis/bundle.py

```diff
@@ -0,0 +1,267 @@
+            for i, sim in enumerate(bundle):
+                sim.universe.trajectory.add_auxiliary(auxname=auxname, 
+                                                      auxdata=auxvals[i], **args) 
+
+    ## Add any metadata
+    data = data if data is not None else {}
+    for key, values in data.items():
+        if (isinstance(values, (list, np.ndarray)) and 
+           len(values) != len(simulations)):
+            raise ValueError("Number of simulations ({}) and number of values for"
+                             " metadata {} ({}) do not match".format(
+                                            len(simulations), key, len(values)))
+        bundle.categories[key] = values
+
+    return bundle
+
+
+def add_to_bundle(bundle=None, simulation=None, name=None, index=None, 
+                  topology=None, univ_kwargs=None):
+    """ Insert *simulation* into *bundle* at the position *index* and return new
+    bundle.
+
+    If *bundle* is None, a new bundle is created containing just *simulation*
+    and returned. 
+
+    *simulation* may be [[ ... see above, or list here as well? ]]
+
+    Parameters
+    ----------
+    bundle : datreant Bundle object, optional
+        bundle to insert simulation into
+    simulation :
+        simulation to add; [[ list possible types? ]]
+    name : str
+        name for the simulation
+    index : int, optional
+        position in which to insert *simulation*. If *None* (default), the 
+        simulation will be added in the last position.
+    topology : str, optional
+        filename for corresponding topology if simulation is provided as a 
+        trajectory filename
+    univ_kwargs : dict, optional
+        dictionary of additional arguments to pass to Universe.
+
+
+    Returns
+    -------
+    bundle : datreant Bundle object
+        updated bundle
+    """
+    new_sim = _make_sim(simulation, name=name, topology=topology, 
+                        univ_kwargs=univ_kwargs)
+    if bundle is None:
+        # make a new bundle and return
+        # TODO - should we warn this is what we're doing?
+        return dtr.Bundle(simulation)
+    elif isinstance(bundle, dtr.Bundle):
+        if index is None:
+            bundle.add(new_sim)
+        else:
+            sims = [sim for sim in bundle]
+            sims.insert(index, new_sim)
+            bundle = dtr.Bundle(sims)
+        return bundle
+    else:
+         raise TypeError("Invalid bundle {}".format(bundle))
+
+
+def _make_sim(simulation, name=None, topology=None, univ_kwargs=None):
+    """ Return a MDSynthesis Sim for *simulation* with name *name*. """
+    univ_kwargs = univ_kwargs or {}
+    if isinstance(simulation, Sim):
+        # TODO - warn that name will be ignored?
+        return simulation
+    # TODO - allow to just provide name of existing saved Sim
+
+    # check we have a valid *name* and make a new Sim
+    if not isinstance(name, str):
```

Comment:
*name* could also be of type `unicode`. Use `isinstance(name, six.string_types)` instead.